frontend/app/screens/Sessions/index.js

```diff
@@ -0,0 +1,200 @@
+// @flow
+
+import React from 'react';
+import Helmet from 'react-helmet';
+import Header from 'Header';
+import styles from 'Sessions/styles.css';
+import withSessionQuery from 'Sessions/withSessionsQuery';
+import type { SessionT, SkillLevelT, TrackT } from 'types';
+import { cleanHtml, searchArr, titleToLink } from 'utils';
+import compose from 'recompose/compose';
+import withState from 'recompose/withState';
+import withHandlers from 'recompose/withHandlers';
+import lifecycle from 'recompose/lifecycle';
+import withProps from 'recompose/withProps';
+import Link from 'AsyncLink';
+
+type SessionExtendedT = SessionT & {
+  visibility: boolean,
+};
+
+const withLogic = compose(
+  withState('sessions', 'setSessions', ({ sessions }) => sessions),
+  withState('skillLevel', 'setSkillLevel', null),
+  withState('track', 'setTrack', null),
+  withState('textSearchInput', 'setSearchText', ''),
+  withHandlers({
+    filterBySkillLevel: ({
+      setSessions,
+      sessions,
+      setSkillLevel,
+    }) => skillLevel => {
+      setSkillLevel(skillLevel);
+      setSessions(
+        sessions.map(el => {
+          if (el.skillLevel === skillLevel) {
+            return { ...el, visibility: true };
+          }
+          return { ...el, visibility: false };
+        }),
+      );
+    },
+    filterByTrack: ({ setSessions, sessions, setTrack }) => track => {
+      setTrack(track);
+      setSessions(
+        sessions.map(el => {
+          if (el.track === track) {
+            return { ...el, visibility: true };
+          }
+          return { ...el, visibility: false };
+        }),
+      );
+    },
+    filterByText: ({ setSessions, sessions, setSearchText }) => ({
+      target: { value },
+    }) => {
+      setSearchText(value);
+      if (!value) {
+        setSessions(sessions.map(el => ({ ...el, visibility: true })));
+      }
+      const sessionsFilteredBySearchResults = searchArr(
+        value,
+        sessions,
+      ).map((isMatch, eachInd) => ({
+        ...sessions[eachInd],
+        visibility: isMatch,
+      }));
+      setSessions(sessionsFilteredBySearchResults);
+    },
+    resetAllFilters: ({
+      sessions,
+      setSessions,
+      setSkillLevel,
+      setTrack,
+      setSearchText,
+    }) => () => {
+      setSessions(sessions.map(el => ({ ...el, visibility: true })));
+      setSkillLevel(null);
+      setTrack(null);
+      setSearchText('');
+    },
+  }),
+  withProps(() => ({
+    tracks: [
```

Comment:
These aren't the final tracks and/or skill levels. Would be great to not hardcode these. Easy way would be to generate from the list from the sessions. If we use graphql paging, then there would be no guarantee that every options is surfaced.

Best way is to use graphql to query the taxonomy terms (that's why I said to to use taxonomy for these fields).